test/test_compile_part_4.py

```diff
@@ -292,6 +292,22 @@ def test_compile_with_esp8266_bundled_libraries(run_command, data_dir, copy_sket
     assert "\n".join(expected_output) not in res.stdout
 
 
+def test_generate_compile_commands_json_with_esp32(run_command, data_dir, copy_sketch):
+    # https://github.com/arduino/arduino-cli/issues/1547
+    assert run_command(["update"])
+
+    # Update index with esp32 core and install it
+    url = "https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json"
```

Comment:
In that case, we should also cache the tools and the cores that may be removed in the future...

BTW I would not care now, we never had this problem before. IMHO we can just pin the version as suggested by @per1234, if the index.json or the core is removed it will make the test fail and we will take action only then.